src/components/Form.js

```diff
@@ -1,9 +1,54 @@
-const Form = () => (
-  <form>
-    <input type="text" name="title" placeholder="Book title" />
-    <input type="text" name="author" placeholder="Book author" />
-    <button type="button">Add Book</button>
-  </form>
-);
+import { useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { add } from '../redux/books/booksSlice';
+
+const Form = () => {
+  const books = useSelector((state) => state.books);
+  const getId = (books) => {
+    const ids = books.map((element) => element.item_id);
+    const max = Math.max(...ids);
+    return max + 1;
+  };
+  const dispatch = useDispatch();
+  const [title, setTitle] = useState('');
+  const [author, setAuthor] = useState('');
+  const submitHandler = (e) => {
+    e.preventDefault();
+    const newBook = {
+      item_id: getId(books),
+      title,
+      author,
+      category: 'fiction',
+    };
+    dispatch(add(newBook));
+  };
```

Comment:
- [OPTIONAL] After submitting the form data for each book, it would be nice to clear the data. It would improve the user experience. 👍 😎